WordPress/openverse
# Add an analytics event for search results count

## Fixes



Fixes #4561 by @obulat 

## Description


This PR introduces a new analytics event, `GET_SEARCH_RESULTS`, that logs the number of results returned for each user search query for each media type (image and audio). By adding `GET_SEARCH_RESULTS`, we gain insight into queries that yield few or no results, enabling us to better understand and address potential gaps in openverse search capabilities.

As was advised in issue #4561 a `GET_SEARCH_RESULTS` event was placed near the existing `SUBMIT_SEARCH` event. The very sending of the event was implemented in the `fetchSingleMediaType` function before we return mediaCount, if page is 1 (so, it fires once per a search term, and does not fire when user clicks on Load more)
```typescript
if (page == 1) {
    $sendCustomEvent("GET_SEARCH_RESULTS", {
         mediaType: mediaType,
         query: queryParams.q,
         resultsCount: mediaCount
    })
}
```

## Testing Instructions

Project setup and initialisation was done according to the [General setup guide](https://docs.openverse.org/general/general_setup.html), but despite this, for some reason analytical events were not sent to the Plausible server (http://0.0.0.0:50288/localhost), as it was unreachable for none of them. Even though it was started, initialised and there were no visible errors in logs. So, this didn't give the ability to test and see the stats directly in the form of a chart etc on Plausible server.
![Screenshot_1](https://github.com/user-attachments/assets/39c62070-4767-4868-a3e0-c2bfec5b972f)

Despite this operability can be tested using the developer tools in the browser. The following image shows that events are successfully sent as intended without any visible bugs.
![Screenshot_2](https://github.com/user-attachments/assets/80137151-9e55-4b79-91ca-b12e32708d78)


## Checklist


      PRs) or the media properties generator (`ov just catalog/generate-docs media-props`
      for the catalog or `ov just api/generate-docs` for the API) where applicable.

[best_practices]:
  https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin


<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>
